### PR TITLE
Adds detail url to cache DB (allows to open torrent details in manual search)

### PR DESCRIPTION
**Pull request by duramato**
#TODO:

- [ ] RARBG: need to improve unshorten detail info to only get the redirect URL and NOT load the URL

Providers that need to be done:

- [x] HD4Free 
- [x] TorrentLeech (and adds pubdate)
- [X] HoundDawgs
- [x] HDTorrents
- [X] GFTracker 
- [x] HDSpace
- [x] Speedcd (and adds pubdate and fixes login)
- [ ] TorrentProject 
- [x] TvChaosUK (and adds pubdate)
- [X] Pretome (and adds pubdate)
- [X] TNTVillage
- [X] BTN (and adds pubdate)
- [x] Torrentz2
- [x] Torrenting
- [x] SDBits
- [X] BITHDTV
- [X] Danishbits
- [X] HDBits
- [X] AlphaRatio
- [x] LimeTorrents
- [x] SceneTime (and adds pubdate)
- [x] Nebulance
- [X] TorrentDay 
- [X] MoreThanTV 
- [X] TorrentBytes 
- [ ] ThePirateBay
- [x] Rarbg 
- [ ] Zooqle
- [x] IPTorrents
- [ ] Newpct 
- [ ] BitCannon 
- [ ] TokyoToshokan 
- [ ] AnimeBytes 
- [x] EliteTorrent 
- [ ] Xthor
- [ ] Shazbat.tv 
- [ ] ABNormal
- [ ] Norbits
- [ ] AnimeTorrents
- [ ] HorribleSubs
- [ ] AniDex
- [ ] ShanaProject
- [ ] Nyaa
- [ ] Torrent9
- [ ] Yggtorrent
- [ ] HeBits
- [ ] EliteTracker
- [ ] ArcheTorrent

Implements: https://github.com/pymedusa/Medusa/issues/626

### CODE REVIEW

File: medusa/tv/cache.py [hunk 1]
```diff
@@ -591,6 +597,7 @@ def find_needed_episodes(self, episode, forced_search=False,
             result.size = cur_result[b'size']
             result.pubdate = cur_result[b'pubdate']
             result.proper_tags = cur_result[b'proper_tags'].split('|') if cur_result[b'proper_tags'] else ''
+            resilt.detail_url = cur_result[b'detail_url']
```
Comment (labrys) [inline medusa/tv/cache.py]: `result.detail_url`

File: medusa/providers/torrent/json/rarbg.py [hunk 2]
```diff
@@ -195,13 +195,18 @@ def parse(self, data, mode):
                 pubdate = row.pop('pubdate')
                 pubdate = parser.parse(pubdate, fuzzy=True)
 
+                shorten_detail_url = row.pop('info_page')
+                response = self.get_url(shorten_detail_url)
+                detail_url = response.url
```
Comment (fernandog) [inline medusa/providers/torrent/json/rarbg.py]: need to improve this to only get the redirect URL and NOT load the URL

**Comment by fernandog** [comment]
@pymedusa/developers are we going to continue this PR?

The only issue I see with this PR is RARBG that shorten the url and we need to unshorten before save to db

**Comment by duramato** [comment]
The main thing missing now is showing it in the snatchSelection

**Comment by medariox** [comment]
I think there is no point in keeping this open. Just too much work to keep it working for every provider, sorry.

**Pull request closed without merging**